app/src/main/java/framgia/com/ichat/data/source/local/AuthenticationLocalDataSource.java

```diff
@@ -0,0 +1,52 @@
+package framgia.com.ichat.data.source.local;
+
+import android.content.SharedPreferences;
+
+import framgia.com.ichat.data.source.AuthenticationDataSource;
+import framgia.com.ichat.data.source.CallBack;
+import framgia.com.ichat.screen.login.LoginKey;
+
+public class AuthenticationLocalDataSource implements AuthenticationDataSource.Local {
+    private SharedPreferences mSharedPreferences;
+
+    public AuthenticationLocalDataSource(SharedPreferences sharedPreferences) {
+        this.mSharedPreferences = sharedPreferences;
+    }
+
+    @Override
+    public void saveUserInformation(boolean isRemember, String email, String password) {
```

Comment:
hàm này e nên tách  ra 
```suggestion
    public void saveUserInformation(String email, String password) {
```

nếu isRemember thì mới gọi hàm saveUserInformation